src/components/auth/signup/buyer-form/SignUpBuyerForm.jsx

```diff
@@ -1,119 +1,81 @@
-import { useState, useEffect, useCallback, use } from 'react'
+import { useState } from 'react'
+import { useAuthStore } from '../../store/AuthStore'
+import {
+  validateEmail,
+  validatePassword,
+  statuses as validationStatuses,
+} from '~/util/ValidationUtil'
 import styles from './SignUpBuyerForm.module.scss'
 
-export default function SignUpBuyerForm({ errors, setErrors, setFormData }) {
-  const [email, setEmail] = useState('')
-  const [password, setPassword] = useState('')
-
-  const validateEmail = useCallback(() => {
-    const pattern = /^[A-Za-z0-9\._%+\-]+@[A-Za-z0-9\.\-]+\.[A-Za-z]{2,}$/
-
-    if (pattern.test(email)) {
-      setErrors((errors) => ({
-        ...errors,
-        email: { valid: true },
-      }))
-
-      return
-    }
-
-    setErrors((errors) => ({
-      ...errors,
-      email: {
-        valid: false,
-        errorMessage: 'Вкажіть коректний email',
-      },
-    }))
-  }, [email])
-
-  const validatePassword = useCallback(() => {
-    const pattern = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{9,}$/
-    if (pattern.test(password)) {
-      setErrors((errors) => ({
-        ...errors,
-        password: { valid: true },
-      }))
-
-      return
-    }
-
-    setErrors((errors) => ({
-      ...errors,
-      password: {
-        valid: false,
-        errorMessage: 'Від 8 буквенних та 1 числовий символи',
-      },
-    }))
-  }, [password])
-
-  const handleEmailChange = (event) => {
-    const newEmail = event.target.value
-    setEmail(newEmail)
+export default function SignUpBuyerForm() {
+  const { email, setEmail } = useAuthStore()
+  const { password, setPassword } = useAuthStore()
+  const { errors, addError, removeError } = useAuthStore()
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false)
 
-    setFormData((prevFormData) => ({
-      ...prevFormData,
-      email: newEmail,
-    }))
+  const handleEyeButton = (e) => {
+    e.preventDefault() // TODO: do we need this ?
+    setIsPasswordVisible((prev) => !prev)
   }
 
-  const handlePasswordChange = (event) => {
-    const newPassword = event.target.value
-    setPassword(newPassword)
-
-    setFormData((prevFormData) => ({
-      ...prevFormData,
-      password: newPassword,
-    }))
+  const handleEmailValidation = (e) => {
+    const { status, error } = validateEmail(e.target.value)
+    if (status === validationStatuses.FAIL) {
+      addError({ email: [error] })
```

Comment:
`[error]` should be replaced with just `error`
```jsx
addError({email: error})
```
`email: error` then will be used as plain string below in 
```jsx
{errors?.email && <div className={styles.errors}>{errors.email}</div>}
```